### PR TITLE
[HOPSWORKS-703] add documentation for plotting on hops

### PR DESCRIPTION
**Pull request by Limmen**

### CODE REVIEW

**Comment by Limmen** [comment]
@robzor92 @tkakantousis added a Jupyter documentation page that explains the basics about the notebooks. E.g:

- What pyspark is
- How spark session is created automatically when starting notebooks
- The difference between a pyspark notebook and a regular python notebook
- How to do plotting

Also added a link from the tensorflow documentation to the Jupyter page, so the tensorflow page can focus on the tensorflow/deep learning specifics on Jupyter, and the Jupyter page can just describe the basic information about how a notebook works. I think this make more sense than to have all information under the "tensorflow on hops" page, I.e the ToC for Hopsworks user guide now looks like this:

pr

File: docs/user_guide/hopsworks/jupyter.rst [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+    :alt: Open the Jupyter service on Hopsworks
+    :target: `jupyter1.png`_
+    :align: center
+    :figclass: align-center
+
+    Open the Jupyter service on Hopsworks
+
+**Start a Jupyter notebook server**
+
+When you start a Jupyter notebook server you have the possibility to specify Spark properties of the notebooks you will create on the server. Hopsworks provides *Machine learning as a service* with tensorflow and spark, supporting distirbuted training, parallel experiments, hyperparameter tuning, and model serving (HopsML). If you are doing machine learning on hops you probably want to select the notebook servers "Experiment", "Parallel Experiment" or "Distributed training" as shown in the figure below. For more information about *HopsML*, see the hops-tensorflow_ documentation. For general purpose notebooks, select the type "Spark (Static)" or "Spark (Dynamic)".
```
Comment (tkakantousis) [inline docs/user_guide/hopsworks/jupyter.rst]: @Limmen *Machine Learning-as-a-Service*
TensorFlow, Spark

File: docs/user_guide/hopsworks/jupyter.rst [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+    :align: center
+    :figclass: align-center
+
+    Start a Jupyter notebook server
+
+**Jupyter + Spark on Hopsworks**
+
+As a user, you will just interact with the Jupyter notebooks, but if you are interested about the technologies behind it, it is explained below.
+
+When using Jupyter on Hopsworks, a library called `sparkmagic`_ is used to interact with the Hops (hadoop) cluster. When you create a Jupyter notebook on Hopsworks, you first select a **kernel**. A kernel is simply a program that executes the code that you have in the Jupyter cells, you can think of it as a REPL-backend to your jupyter notebook that acts as a frontend.
```
Comment (tkakantousis) [inline docs/user_guide/hopsworks/jupyter.rst]: @Limmen Remove the `(hadoop)`

File: docs/user_guide/hopsworks/jupyter.rst [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+
+Sparkmagic works with a remote REST server for Spark, called `livy`_, running inside the Hops cluster. Livy is an interface that Jupyter-on-Hopsworks use to interact with the Hops cluster. When you run Jupyter cells using the pyspark kernel, the kernel will automatically send commands to livy in the background for executing the commands on the cluster. Thus, the work that happens in the background when you run a Jupyter cell is as follows:
```
Comment (tkakantousis) [inline docs/user_guide/hopsworks/jupyter.rst]: @Limmen `uses`

File: docs/user_guide/hopsworks/jupyter.rst [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+
+- The code in the cell will first go to the kernel.
+- Next, the kernel kernel sends the code as a HTTP REST request to livy.
+- When receiving the REST request, livy executes the code on the Spark driver in the cluster.
+- If the code is regular python/scala/R code, it will be run inside a python/scala/R interpreter on the Spark driver.
```
Comment (tkakantousis) [inline docs/user_guide/hopsworks/jupyter.rst]: @Limmen `will be executed` or `will run`

File: docs/user_guide/hopsworks/jupyter.rst [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+.. _jupyter3.png: ../../_images/jupyter3.png
+.. figure:: ../../imgs/jupyter3.png
+    :alt: Create a pyspark notebook
+    :target: `jupyter3.png`_
+    :align: center
+    :figclass: align-center
+
+    Create a pyspark notebook
+
+When you execute the first cell in a pyspark notebook, the spark session is automatically created, refering to the Hops cluster.
```
Comment (tkakantousis) [inline docs/user_guide/hopsworks/jupyter.rst]: @Limmen `referring`

File: docs/user_guide/hopsworks/jupyter.rst [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+.. figure:: ../../imgs/jupyter6.png
+    :alt: Starting a spark job from Jupyter
+    :target: `jupyter6.png`_
+    :align: center
+    :figclass: align-center
+
+    Starting a spark job from Jupyter
+
+
+When you execute a cell in Jupyter that starts a Spark job, you can go back to the Hopsworks-Jupyter-UI and you will se that a link to the SparkUI for the job have been created.
```
Comment (tkakantousis) [inline docs/user_guide/hopsworks/jupyter.rst]: @Limmen `that has been created`

File: docs/user_guide/hopsworks/jupyter.rst [hunk 7]
```diff
@@ -0,0 +1,227 @@
...
+    :alt: Creating a spark dataframe
+    :target: `jupyter8.png`_
+    :align: center
+    :figclass: align-center
+
+    Creating a spark dataframe
+
+**Step 2 : Download the Spark Dataframe to a local Pandas Dataframe using %%sql or %%spark**:
+
+Note: you should **not** try to download large spark dataframes for plotting. When you plot a dataframe, the entire dataframe must fit into memory, so add the flag `--maxrows x` to limit the dataframe size when you download it to the local Jupyter server for plotting.
```
Comment (tkakantousis) [inline docs/user_guide/hopsworks/jupyter.rst]: @Limmen We can stress out this point even more by bolding the `Note` as well or something like that

File: docs/user_guide/hopsworks/jupyter.rst [hunk 8]
```diff
@@ -0,0 +1,227 @@
...
+    :align: center
+    :figclass: align-center
+
+    Plot a local pandas dataframe using matplotlib and the magic %%local
+
+
+Want to Learn More?
+-------------------
+
+We have provided a large number of example notebooks, available here_, upload them to your project on Hopsworks and try it out! You can also have a look at HopsML (hops-tensorflow_), which enables large-scale distributed deep learning on Hops.
```
Comment (tkakantousis) [inline docs/user_guide/hopsworks/jupyter.rst]: @Limmen maybe we can also mention that these notebooks are already available as part of the Deep Learning tour?

File: docs/user_guide/hopsworks/jupyter.rst (hunk 1 above)
Comment (Limmen) [reply docs/user_guide/hopsworks/jupyter.rst]: Fixed

File: docs/user_guide/hopsworks/jupyter.rst [hunk 9]
```diff
@@ -0,0 +1,227 @@
...
+    :alt: Start a Jupyter notebook server
+    :target: `jupyter2.png`_
+    :align: center
+    :figclass: align-center
+
+    Start a Jupyter notebook server
+
+**Jupyter + Spark on Hopsworks**
+
+As a user, you will just interact with the Jupyter notebooks, but if you are interested about the technologies behind it, it is explained below.
```
[the remark this answers is not in the document]
Comment (Limmen) [reply docs/user_guide/hopsworks/jupyter.rst]: Fixed

File: docs/user_guide/hopsworks/jupyter.rst (hunk 2 above)
Comment (Limmen) [reply docs/user_guide/hopsworks/jupyter.rst]: Fixed

File: docs/user_guide/hopsworks/jupyter.rst (hunk 3 above)
Comment (Limmen) [reply docs/user_guide/hopsworks/jupyter.rst]: Fixed

File: docs/user_guide/hopsworks/jupyter.rst (hunk 4 above)
Comment (Limmen) [reply docs/user_guide/hopsworks/jupyter.rst]: Fixed

File: docs/user_guide/hopsworks/jupyter.rst (hunk 5 above)
Comment (Limmen) [reply docs/user_guide/hopsworks/jupyter.rst]: Fixed

File: docs/user_guide/hopsworks/jupyter.rst (hunk 6 above)
Comment (Limmen) [reply docs/user_guide/hopsworks/jupyter.rst]: Fixed

File: docs/user_guide/hopsworks/jupyter.rst (hunk 7 above)
Comment (Limmen) [reply docs/user_guide/hopsworks/jupyter.rst]: Agree, fixed

File: docs/user_guide/hopsworks/jupyter.rst (hunk 8 above)
Comment (Limmen) [reply docs/user_guide/hopsworks/jupyter.rst]: Agree, fixed

**Pull request merged**